packages/driver/cypress/e2e/e2e/origin/snapshots.cy.ts

```diff
@@ -2,107 +2,153 @@
 import '../../../support/utils'
 
 describe('cy.origin - snapshots', { browser: '!webkit' }, () => {
-  const findLog = (logMap: Map<string, any>, displayName: string, url: string) => {
-    return Array.from(logMap.values()).find((log: any) => {
-      const props = log.get()
+  it('does not create snapshots after the document has unloaded and the AUT has navigated cross-origin', () => {
+    cy.visit('/fixtures/generic.html')
+    cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
+    cy.then(() => {
+      const snapshot = cy.createSnapshot()
 
-      return props.displayName === displayName && (props?.consoleProps?.URL === url || props?.consoleProps()?.URL === url)
+      expect(snapshot).to.be.null
     })
-  }
-  let logs: Map<string, any>
+  })
+
+  it('takes snapshots from the secondary origin even after the primary AUT has been unloaded from state', () => {
+    const findLog = (logMap: Map<string, any>, selector) => {
+      return Array.from(logMap.values()).find((log: any) => {
+        const props = log.get()
 
-  beforeEach(() => {
-    logs = new Map()
+        return (props?.consoleProps?.Selector === selector)
+      })
+    }
+    let logs: Map<string, any> = new Map()
 
     cy.on('log:changed', (attrs, log) => {
       logs.set(attrs.id, log)
     })
 
-    cy.fixture('foo.bar.baz.json').then((fooBarBaz) => {
-      cy.intercept('GET', '/foo.bar.baz.json', { body: fooBarBaz }).as('fooBarBaz')
-    })
-
     cy.visit('/fixtures/primary-origin.html')
     cy.get('a[data-cy="xhr-fetch-requests-onload"]').click()
-  })
 
-  // TODO: the xhr event is showing up twice in the log, which is wrong and causing flake. skipping until: https://github.com/cypress-io/cypress/issues/23840 is addressed.
-  it.skip('verifies XHR requests made while a secondary origin is active eventually update with snapshots of the secondary origin', () => {
     cy.origin('http://www.foobar.com:3500', () => {
-      // need to set isInteractive in the spec bridge in order to take xhr snapshots in run mode, similar to how isInteractive is set within support/defaults.js
-      // @ts-ignore
-      Cypress.config('isInteractive', true)
-      cy.visit('http://www.foobar.com:3500/fixtures/xhr-fetch-requests.html')
-      cy.get(`[data-cy="assertion-header"]`).should('exist')
-      cy.wait('@fooBarBaz')
+      cy.get(`[data-cy="assertion-header"]`)
     })
 
     cy.shouldWithTimeout(() => {
-      const xhrLogFromSecondaryOrigin = findLog(logs, 'xhr', 'http://localhost:3500/foo.bar.baz.json')?.get()
+      const getLogFromSecondaryOrigin = findLog(logs, '[data-cy="assertion-header"]')?.get()
 
-      expect(xhrLogFromSecondaryOrigin).to.not.be.undefined
+      expect(getLogFromSecondaryOrigin).to.not.be.undefined
```

Comment:
```suggestion
      expect(getLogFromSecondaryOrigin).to.be.defined
```

Reply:
I am not sure why but I cannot get this type to show up in a chai assertion. I don't think it is actually valid 🙁

Reply:
Yeah, doesn't seem to be a [bundled chai assertion](https://www.chaijs.com/api/bdd/), but we add on to chai with various libs, so it's possible it ends up as an assertion somehow, since we do use `to.be.defined` in a few other tests. I think the "correct" assertion is `.to.exist`.

Reply:
updated to `to.exist` in 17d37f56ec